### ISSUE TITLE
Command History Picker: Sorting order when searching

### ISSUE THREAD

**Issue by tuxflo** [nvim-telescope/telescope.nvim#1118]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I'm using the `command_history` Picker. Without searching anything the results are sorted correctly by when they where last used (https://github.com/nvim-telescope/telescope.nvim/issues/150) but when typing some letters the time based order is gone.

**Describe the solution you'd like**
It would be great if the results where ordered by the last usage time even if they where filtered by some keywords/letters.

**Describe alternatives you've considered**
I took a look on the "Frecency" plugin, but as far as I can see this is only for files, not for commands.

**Additional context**
In the screenshots the problem is visible, without searching the `...('dapui').toggle` command was executed recently, if I search for "toggle" it appears way up in the list.
![image](https://user-images.githubusercontent.com/2816893/129494598-2979283f-732d-40f7-b53f-7b90442c48f7.png)
![image](https://user-images.githubusercontent.com/2816893/129494604-13857b4c-3335-47b5-bfab-d05f23a3f866.png)

**Comment by Conni2461** [comment]
You can do this by picking a sorter that treats each match the same, so returns the same score. For example the `get_substr_matcher`. In this case you will lose fuzzy matching. Thats also what `frecency` does and this sorters is written by the same person, for this usecase.

```lua
lua require("telescope.builtin").command_history({sorter = require("telescope.sorters").get_substr_matcher()})
```

The thing right now is that we are currently working towards changing the insert for elements with the same score in #987 by prefering the shorter item if the scores are the same. That improves sorting with `fzf-native` for items with the same score but will break this usecase. I will look for a solution, hopefully before it gets merged :laughing:

**Comment by mcchrish** [comment]
Hi I'm new to telescope but according to this https://github.com/nvim-telescope/telescope.nvim/issues/150 it should be sorting the history starting from the most recent by default. However mine looks to be sorting by length:
<img width="912" alt="Screen Shot 2021-08-21 at 5 16 17 AM" src="https://user-images.githubusercontent.com/7200153/130294404-ff000a87-7d82-43dc-a10b-59a2a273d013.png">
Running just plain `:Telescope command_history`.

@Conni2461 that code snippet however is sorting the results correctly.

Do you think it's currently broken by the async changes?

**Comment by Conni2461** [comment]
This was due to the async changes and has been resolved for quite some time.

We already just merged #1401 which allows a custom `tiebreak`. Closing
